### PR TITLE
Add some chapters to Book

### PR DESCRIPTION
**Pull request by Tristan1900**
**Change log description**  
Add examples to Pravega Rust client and add some chapters to the Pravega Rust client book.
FIx a minor test error
 
**Purpose of the change**

### CODE REVIEW

**Review by shrids — APPROVED** [review]
LGTM

File: src/event_stream_writer.rs [hunk 1]
```diff
@@ -20,8 +20,50 @@ use crate::reactor::event::{Incoming, PendingEvent};
 use tracing::info_span;
 use tracing_futures::Instrument;
 
-/// EventStreamWriter contains a writer id and a mpsc sender which is used to send Event
-/// to the StreamWriter
+/// Writes events exactly once to a given stream with backpressure.
```
Comment (tkaitchuck) [inline src/event_stream_writer.rs:23]: Move discussion of backpreasure to the write method as there are more details.

File: src/event_stream_writer.rs [hunk 2]
```diff
@@ -56,6 +112,16 @@ impl EventStreamWriter {
         }
     }
 
+    /// Writes an event with a routing key.
+    ///
+    /// Write has a backpressure mechanism. Internally, it uses [`Channel`] to send event to
+    /// Reactor for processing. [`Channel`] can has a limited [`capacity`], when its capacity
+    /// is reached, any further write will be blocked until enough space has been freed in the [`Channel`].
```
Comment (tkaitchuck) [inline src/event_stream_writer.rs:119]: I don't think "Blocked" is the right word as the method is async.

File: src/event_stream_writer.rs [hunk 3]
```diff
@@ -45,6 +89,18 @@ impl EventStreamWriter {
         }
     }
 
+    /// Writes an event without routing key.
+    ///
+    /// A random routing key will be generated in this case.
+    ///
+    /// Write has a backpressure mechanism. Internally, it uses [`Channel`] to send event to
+    /// Reactor for processing. [`Channel`] can has a limited [`capacity`], when its capacity
+    /// is reached, any further write will be blocked until enough space has been freed in the [`Channel`].
```
Comment (tkaitchuck) [inline src/event_stream_writer.rs:98]: same here

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**